### PR TITLE
Default network costs and load balancer cost, Has a provision to have configs in opencost be second line of values in Closed source and fixes a bug

### PR DESCRIPTION
**Pull request by avrodrigues5**
## What does this PR change?
* Add default network costs and load balancer costs in Alibaba provider
* Noticed that in all provider even thou we pass /models in closed source the value used is defaultPricing values so added the config value as second line of information if user doesnt give their own values using /updateConfigByKey end point. This will only be in effect if user deleted the existing config else it continue to be default pricing itself. I dont want to be destructive.
* Fixes a bug wherein alibaba service key and secret were set to empty if there were passed in default config.

## Does this PR relate to any other PRs?
* None

## How will this PR impact users?
* Give network costs and loadbalancer cost in their alibaba environment

## Does this PR address any GitHub or Zendesk issues?
* Closes ... [1251](https://github.com/kubecost/cost-analyzer-helm-chart/issues/2151)

## How was this PR tested?
* Tested on alibaba cluster to show the Network costs and LB costs
Screen Shot 2023-04-27 at 5 29 15 PM

* Prior to this fix if secret and key were passed in end point the overwrite of it would be empty and the REST end point would fail now it shows the node pricing and PV pricing... Might solve [1250](https://github.com/kubecost/features-bugs/issues/74) if user was doing the same.

* Verified that the default pricing gets created with alibaba.json in config if current written with default pricing is removed.

Screen Shot 2023-04-27 at 5 33 18 PM
Screen Shot 2023-04-27 at 5 33 32 PM
 

## Does this PR require changes to documentation?
* Maybe requesting user to edit the network pricing for their region and zone give by alibaba documentation would be great 
https://www.alibabacloud.com/help/en/cloud-data-transmission/latest/cross-region-data-transfers
* Alibaba's load balancer is all divided into 4 types and each have different prices 
https://www.alibabacloud.com/product/server-load-balancer
Defaulted to Classic load balancer https://www.alibabacloud.com/help/en/server-load-balancer/latest/pay-as-you-go

I have documented this in code but would be great to have documentation around it!

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?
* v1.103 but not release blocking can go to v1.104

### CODE REVIEW

**Comment by avrodrigues5** [comment]
@mbolt35 When you have time please review my PR addressing all the key points related to network costs and loadbalancer cost as discussed
!

File: pkg/cloud/providerconfig.go [hunk 1]
```diff
@@ -91,6 +95,15 @@ func (pc *ProviderConfig) loadConfig(writeIfNotExists bool) (*CustomPricing, err
...
 		pc.customPricing = DefaultPricing()
+		// If config file is not present use the contents from mount models/ as pricing data
+		// in closed source rather than from from  DefaultPricing as first source of truth.
+		// since most images will already have a mount, to avail this facility user needs to delete the
+		// config file manually from configpath else default pricing still holds good.
+		fileName := filenameInConfigPath(pc.configFile.Path())
+		defaultPricing, err := ReturnPricingFromConfigs(fileName)
+		if err == nil {
+			pc.customPricing = defaultPricing
+		}
```
Comment (mbolt35) [inline pkg/cloud/providerconfig.go:106]: Looks awesome, were you able to verify that this worked well with the other providers?

**Review by mbolt35 — APPROVED** [review]
This looks good to me. I would just make sure that the ProviderConfig updates you made work on other providers as well.

File: pkg/cloud/providerconfig.go (hunk 1 above)
Comment (avrodrigues5) [reply pkg/cloud/providerconfig.go:106]: Verified in GCP and Azure:

Test strategy was: 
1. use rc-6 upload a default price config and then set the new image does it show up with the same sset default - yes it did.
2. when image was set to fixed version the configs mounted on /model showed up in the /var/config as expected.

Spoke to Bolt this looks good and merging the code!

**Pull request merged**